### PR TITLE
[SPARK-49036] Exclude `JUnitAssertionsShouldIncludeMessage/JUnitTestContainsTooManyAsserts` PMD rules and simplify test code

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

Currently, `PMD Source Code Analyzer` fails to detect our test code issues.

This PR aims to simplify assertion test code like the following desirable patterns for better source code analysis.
1. Use better assertion method
```java
- Assertions.assertTrue(constructorArgs.get(conf).get(2) instanceof PythonMainAppResource);
+ assertInstanceOf(PythonMainAppResource.class, constructorArgs.get(conf).get(2));
```

2. Avoid repeating `Assertions.` in the test code by using static import statements to help PMD analyze the code easily
```java
- Assertions.assertEquals(generation, 1L);
+ assertEquals(generation, 1L);
```

In addition, according to the existing written test code, we need to exclude the following two rules because we have many rule violation already.
-  `JUnitAssertionsShouldIncludeMessage`
- `JUnitTestContainsTooManyAsserts`

### Why are the changes needed?

To keep the test code in a simple and uniform form via PMC Source Code Analyzer.
- The existing pattern, `Assertions.*` was not covered by PMC Source Code Analyzer properly.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @jiangzho and @viirya

File: config/pmd/ruleset.xml [hunk 1]
```diff
@@ -21,13 +21,12 @@
...
-  <rule ref="category/java/bestpractices.xml"/>
+  <rule ref="category/java/bestpractices.xml">
+    <exclude name="JUnitAssertionsShouldIncludeMessage" />
+    <exclude name="JUnitTestContainsTooManyAsserts" />
+  </rule>
   <rule ref="category/java/security.xml"/>
   <rule ref="rulesets/java/basic.xml"/>
-  <rule ref="category/java/bestpractices.xml/JUnitTestContainsTooManyAsserts">
-    <properties>
-      <property name="maximumAsserts" value="10"/>
```
Comment (dongjoon-hyun) [inline config/pmd/ruleset.xml:29]: This was already broken. So, I removed this.

File: config/pmd/ruleset.xml [hunk 2]
```diff
@@ -21,13 +21,12 @@
   <description>
     Spark Operator Ruleset
   </description>
-  <rule ref="category/java/bestpractices.xml"/>
+  <rule ref="category/java/bestpractices.xml">
+    <exclude name="JUnitAssertionsShouldIncludeMessage" />
```
Comment (dongjoon-hyun) [inline config/pmd/ruleset.xml:25]: This is recommended, but we have many assertions already. We don't need to enforce it.

**Review by viirya — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya . Merged to main.

**Pull request closed without merging**